internal_transfers/actions/src/parse.ts

```diff
@@ -1,63 +1,95 @@
-import { SettlementEvent, TradeEvent, TransferEvent } from "./models";
+import {
+  isSettlementEvent,
+  isTradeEvent,
+  SettlementEvent,
+  TradeEvent,
+  TransferEvent,
+} from "./models";
 import { Log } from "@tenderly/actions";
 import { transferInvolves } from "./utils";
 import { ethers } from "ethers";
 import { abi as SETTLEMENT_ABI } from "@cowprotocol/contracts/lib/contracts/GPv2Settlement.json";
 import { abi as IERC20_ABI } from "@cowprotocol/contracts/lib/contracts/IERC20.json";
 import { address as SETTLEMENT_CONTRACT_ADDRESS } from "@cowprotocol/contracts/deployments/mainnet/GPv2Settlement.json";
 
+const settlementInterface = new ethers.Interface(SETTLEMENT_ABI);
+const erc20Interface = new ethers.Interface(IERC20_ABI);
 export interface ClassifiedEvents {
   trades: TradeEvent[];
   transfers: TransferEvent[];
   settlements: SettlementEvent[];
 }
 
 export function partitionEventLogs(logs: Log[]): ClassifiedEvents {
-  let result: ClassifiedEvents = {
+  const result: ClassifiedEvents = {
     transfers: [],
     trades: [],
     settlements: [],
   };
-  const settlementInterface = new ethers.Interface(SETTLEMENT_ABI);
-  const erc20Interface = new ethers.Interface(IERC20_ABI);
+
   logs.forEach((log, index) => {
     // We are only interested in Transfer Events from erc20 contracts
     // along with Settlement and Trade Events from the Settlement contract
     // All other event emissions can be ignored for our purposes.
-    const transferEventLog = erc20Interface.parseLog(log);
-    const settlementEventLog = settlementInterface.parseLog(log);
-    if (transferEventLog != null && transferEventLog.name === "Transfer") {
-      const { from, to, value } = transferEventLog.args;
-      const transfer: TransferEvent = {
-        token: log.address,
-        from,
-        to,
-        amount: BigInt(value),
-      };
-      // Is a "relevant" transfer (involving settlement contract)
+    const possibleTransfer = tryParseERC20Transfer(log);
+    if (possibleTransfer) {
+      const transfer = possibleTransfer;
```

Comment:
You still reassign to a new const, which is not necessary

Reply:
Good point. Will remove at some point.